### ISSUE TITLE
The Results are Unreproducible

### ISSUE THREAD

**Issue by zhouyuan888888** [JackYu6/HERO_release#6]
Hi, thanks for your interesting work! By using the official code without any modification, I cannot reproduce the results in the paper! 

After running
`python train_mask_transformer_memo_cross_vimo.py --name mtrans_memo_cross_l6_bs64_ep200 --gpu_id 0 \
    --dataset_name vimo --batch_size 64 --max_epoch 200 --vq_name rvq_bs256_finetune_ep10 \
    --milestones 6000 --warm_up_iter 250 --n_layers 6`

The final FID during training is even 1.7991

<img width="2278" height="1090" alt="Image" src="https://github.com/user-attachments/assets/38b96c96-fc3a-4e61-b473-6108551dbc29" />

Could you help me to reproduce the results in the paper?

**Comment by JackYu6** [comment]
Hi! Thanks for your interest.
Usually, the final checkpoint from training is not the optimal one. Following [MoMask](https://github.com/EricGuo5513/momask-codes/tree/main), we evaluate the models saved as `net_best_acc.tar` or `net_best_fid.tar`. You can specify which checkpoint to use with `--which_epoch` when running `python eval_trans_res_memo_cross_vimo.py`.

Before running the evaluation code, please make sure that:
- Your RVQ-VAE loads the pretrained weights before fine-tuning;
- You have finished training all models, including the RVQ-VAE, Masked Transformer, and Residual Transformer.

Training and evaluation on different machines may yield different results, but the differences should not be significant.

**Comment by zhouyuan888888** [comment]
Thanks for your quick reply. I will run the code again!

**Comment by SreehariRajan** [comment]
I didn't get this statement, "Your RVQ-VAE loads the pretrained weights before fine-tuning;", 
Can you explain this?

**Comment by JackYu6** [comment]
> I didn't get this statement, "Your RVQ-VAE loads the pretrained weights before fine-tuning;", Can you explain this?

Please download our checkpoints and make sure that you add `--finetune` when running `python train_vq_vimo.py`. This will initialize the RVQ-VAE with the weights obtained by pretraining on HumanML3D dataset.

**Comment by SreehariRajan** [comment]
Ok, also, what will be the target FID we should aim for VQVAE validation?

**Comment by JackYu6** [comment]
FID: 0.133±0.002, MPJPE: 32.8

**Comment by SreehariRajan** [comment]
When I was trying to fine-tune the VQVAE with another small custom dataset, the FID was increasing from 0.4 to around 3. ANy solution for this ?

**Comment by JackYu6** [comment]
Did you use the mean and std of your custom dataset when training?

**Comment by SreehariRajan** [comment]
Yes

**Comment by JackYu6** [comment]
Perhaps the data distribution in your dataset differs significantly from that of HumanML3D. I recommend conducting pre-training on a larger motion dataset.

**Comment by SreehariRajan** [comment]
For comparison with MDM and MLD, how many epochs were used to train them?

**Comment by SreehariRajan** [comment]
> Perhaps the data distribution in your dataset differs significantly from that of HumanML3D. I recommend conducting pre-training on a larger motion dataset.

If we are doing this, how can we evaluate?  Because those stats will be different compared to the stats of evaluator?

**Comment by JackYu6** [comment]
You can evaluate the models using the same evaluator. Although the values of the results depend on the evaluator used, comparisons across models are relative.
